ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/datalayer/datalayer.js

```diff
@@ -87,39 +95,61 @@
                 }
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.FIELD, formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.FIELD, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('elementHelpShown', function(event){
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.HELP, event.detail.formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.HELP, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('elementErrorShown', function(event){
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: getEventInfo(FormEvents.ERROR, event.detail.formTitle, event.detail.fieldId, dataLayerContent)
+                    eventInfo: getEventInfo(FormEvents.ERROR, event.detail.formTitle, event.detail.fieldId, dataLayerContent, formContainerID)
                 });
             });
 
             bridge.on('submitStart', function(event){
-                let eventInfo = {
-                    eventName : FormEvents.SUBMIT,
-                    fieldTitle,
-                    fieldType,
-                    panelTitle,
-                    formTitle
-                }
-                eventInfo.timeSpentOnForm = ((new Date()).getTime() - startTime) / 1000;
+                containerState[formContainerID].submitted = true;
                 dataLayer.push({
                     event: FASTTRACK_ANALYTICS_EVENT,
-                    eventInfo: eventInfo
+                    eventInfo: {
+                        type: FormEvents.SUBMIT,
```

Comment:
Can you write a test case for the event payload, so that I could quickly analyse the changes

Reply:
Added test for event: https://github.com/adobe/aem-core-forms-components/pull/662/files#diff-66f64f01390ab8bf206c9c13d12586a3dbf0574de7989f46171b16ced196f6b9R72

The schema matches the af2 headless schema: https://git.corp.adobe.com/pages/livecycle/af2-docs/spec/latest/#_event_object

![Screenshot 2023-06-01 at 12 48 55 PM](https://github.com/adobe/aem-core-forms-components/assets/16068034/5e7e79a9-64fe-41be-bbff-683d3440b1eb)

Reply:
Please remove references to git.corp in github repo. Can you delete the comment